### PR TITLE
Property support

### PR DESCRIPTION
**Pull request by atrifan**
* added environment variables support
* added properties support
* don't start gateway without a generated-conf.d
* better supervisor

### CODE REVIEW

**Review by bogdanciuca — APPROVED** [review]
👍

**Review by corinapurcarea — APPROVED** [review]
👍

File: init.sh [hunk 1]
```diff
@@ -77,13 +91,21 @@ if [[ -n "${remote_config}" ]]; then
       echo "   ... but this REMOTE_CONFIG is not supported "
     fi
 fi
-sudo api-gateway-config-supervisor \
+
+api-gateway-config-supervisor \
         --reload-cmd="api-gateway -s reload" \
-        --sync-folder=/etc/api-gateway \
+        --sync-folder=/usr/local/api-gateway/lualib \
```
Comment (ddragosd) [inline init.sh:97]: Using the same sync_cmd as bellow ?

File: api-gateway-config/conf.d/marathon_apis.conf [hunk 2]
```diff
@@ -1,191 +0,0 @@
-#/*
```
Comment (ddragosd) [inline api-gateway-config/conf.d/marathon_apis.conf:1]: Ok to remove it as default, but let’s update README and document how to configure the GW w Marathon. Most of this is already written up, just needs some reorg.

File: api-gateway-config/conf.d/tracking_api.conf [hunk 3]
```diff
@@ -3,7 +3,11 @@
 # This VHost should use a server_name matching the hostname
 
 server {
-  listen 5000 default_server;
+  listen 5000;
+  server_name default_server;
+  access_log /var/log/api-gateway/tracking_access.log;
```
Comment (ddragosd) [inline api-gateway-config/conf.d/tracking_api.conf:8]: Logging to stdout avoids filling up all the disk space and causing an out of space issue in containers.

File: init.sh [hunk 4]
```diff
@@ -45,23 +45,37 @@ function start_zmq_adaptor()
         zmq_adaptor_cmd="${zmq_adaptor_cmd} -d"
     fi
 
-    sudo $zmq_adaptor_cmd >> /dev/stderr &
-    sleep 3s
+    $zmq_adaptor_cmd >> /dev/stderr &
+    sleep 10s
```
Comment (ddragosd) [inline init.sh:49]: do 7s make any difference here ? It accounts for large configs ?

File: init.sh [hunk 5]
```diff
@@ -45,23 +45,37 @@ function start_zmq_adaptor()
...
 }
 # keep the zmq adaptor running using a simple loop
 while true; do zmq_pid=$(ps aux | grep api-gateway-zmq-adaptor | grep -v grep) || ( echo "Restarting api-gateway-zmq-adaptor" && start_zmq_adaptor ); sleep 60; done &
 
 
 echo "Starting api-gateway ..."
+
+was_empty="false"
+
+while ! [ "$(ls -A /etc/api-gateway/generated-conf.d)" ]; do
```
Comment (ddragosd) [inline init.sh:61]: This functionality should be configurable.

**Review by ddragosd — CHANGES_REQUESTED** [review]
Pls review my comments above. Thanks !

**Comment by atrifan** [comment]
@ddragosd this branch will not be merged anytime soon this branch is needed only for building internal apigateway dockers only. Rest assured that if we decide to merge it into master will try to make it as backward compatible as it can be

File: init.sh (hunk 4 above)
Comment (atrifan) [reply init.sh:49]: it accounts for large configs

File: init.sh (hunk 5 above)
Comment (atrifan) [reply init.sh:61]: yes it should be but a known issue is that if the gateway starts and the confs come pooring down the pipe as they are beeing downloaded you will have a massive chunk of reloads.

**Comment by atrifan** [comment]
will close this PR since it's not going to be merged

**Pull request closed without merging**